eclair-core/src/main/scala/fr/acinq/eclair/router/ValidationHandlers.scala

```diff
@@ -0,0 +1,448 @@
+/*
+ * Copyright 2020 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.router
+
+import akka.actor.{ActorContext, ActorRef}
+import akka.event.{DiagnosticLoggingAdapter, LoggingAdapter}
+import fr.acinq.bitcoin.Crypto.PublicKey
+import fr.acinq.bitcoin.Script.{pay2wsh, write}
+import fr.acinq.eclair.blockchain.{UtxoStatus, ValidateRequest, ValidateResult, WatchSpentBasic}
+import fr.acinq.eclair.channel.{BITCOIN_FUNDING_EXTERNAL_CHANNEL_SPENT, LocalChannelDown, LocalChannelUpdate}
+import fr.acinq.eclair.crypto.TransportHandler
+import fr.acinq.eclair.db.NetworkDb
+import fr.acinq.eclair.router.Monitoring.Metrics
+import fr.acinq.eclair.router.Router._
+import fr.acinq.eclair.transactions.Scripts
+import fr.acinq.eclair.wire._
+import fr.acinq.eclair.{Logs, NodeParams, ShortChannelId, TxCoordinates}
+import kamon.Kamon
+
+object ValidationHandlers {
+
+  def sendDecision(peerConnection: ActorRef, decision: GossipDecision): Unit = {
```

Comment:
Doesn't this one need an implicit actor context and to define the `sender`?

Reply:
Great catch. Actually this should be enough:
```suggestion
  def sendDecision(peerConnection: ActorRef, decision: GossipDecision)(implicit sender: ActorRef): Unit = {
```

Reply:
Maybe we should test this. An easy way to test this would be to add a check to existing tests that the `sender` they see if the router?

Reply:
I did, with 5ca43a6, and found another bug 🙃 . I should test all the rejection cases but can't do it rn.

Reply:
Sounds good, I like the current state of the PR.
Once we add some more tests it should be good to go from my point of view.

Reply:
Phew, done with c738659. No new bug found.